hazelcast/src/main/java/com/hazelcast/spi/impl/operationexecutor/slowoperationdetector/SlowOperationDetector.java

```diff
@@ -253,9 +266,8 @@ private void shutdown() {
             detectorThread.interrupt();
             try {
                 detectorThread.join(SLOW_OPERATION_THREAD_MAX_WAIT_TIME_TO_FINISH);
-            } catch (InterruptedException e) {
-                EmptyStatement.ignore(e);
-                // TODO: Interrupt flag is consumed here
+            } catch (InterruptedException ignored) {
+                detectorThread.interrupt();
```

Comment:
You are restoring the interrupt on the wrong thread. You should restore the interrupt on the calling thread if the join fails, not on the detectorThread. 

And can you split it up in a shut down, that signals to the detector thread to kill himself and add a seperate awaitTermination method? This makes it possible to shutdown all services in 1 go.. and then do the actual waiting.